forc-plugins/forc-doc/src/render.rs

```diff
@@ -460,17 +466,228 @@ pub(crate) enum ContextType {
     /// at a later date if need be
     RequiredMethods(Vec<TyTraitFn>),
 }
+impl ContextType {
+    fn as_block_title(&self) -> BlockTitle {
+        match self {
+            ContextType::StructFields(_) => BlockTitle::Fields,
+            ContextType::StorageFields(_) => BlockTitle::Fields,
+            ContextType::EnumVariants(_) => BlockTitle::Variants,
+            ContextType::RequiredMethods(_) => BlockTitle::RequiredMethods,
+        }
+    }
+}
 #[derive(Clone)]
+pub(crate) struct Context {
```

Comment:
It might be useful to have a comment explaining what context we're talking about here, or maybe a better type name - the name context can certainly be useful (e.g. when providing context to a traversal or something along these lines), but can be a bit ambiguous sometimes too.